meshtastic/mesh_interface.py

```diff
@@ -483,6 +483,60 @@ def sendAlert(
             priority=mesh_pb2.MeshPacket.Priority.ALERT
         )
 
+    def request_user_info(
+        self,
+        destinationId: Union[int, str],
+        wantResponse: bool = True,
+        channelIndex: int = 0,
+    ) -> mesh_pb2.MeshPacket:
+        """Request user information from another node by sending our own user info.
+        The remote node will respond with their user info when they receive this request.
+
+        Arguments:
+            destinationId {nodeId or nodeNum} -- The node to request info from
+
+        Keyword Arguments:
+            wantResponse {bool} -- Whether to request a response with the target's user info (default: True)
+            channelIndex {int} -- The channel to use for this request (default: 0)
+
+        Returns:
+            The sent packet. The id field will be populated and can be used to track responses.
+        """
+        # Get our node's user info to send in the request
+        my_node_info = self.getMyNodeInfo()
+        logger.debug(f"Local node info: {my_node_info}")
+        
+        if my_node_info is None or "user" not in my_node_info:
+            raise MeshInterface.MeshInterfaceError("Could not get local node user info")
+
+        # Create a User message with our info
+        user = mesh_pb2.User()
+        node_user = my_node_info["user"]
+        logger.debug(f"Local user info to send: {node_user}")
+        
+        # Copy fields from our node's user info, matching firmware behavior
+        user.id = node_user.get("id", "")  # Set to nodeDB->getNodeId() in firmware
+        user.long_name = node_user.get("longName", "")
+        user.short_name = node_user.get("shortName", "")        
+        user.hw_model = node_user.get("hwModel", 0)
+        user.is_licensed = node_user.get("is_licensed", False)
+        user.role = node_user.get("role", 0)
+        
+        # Handle public key - firmware strips it if node is licensed
+        if "public_key" in node_user and not user.is_licensed:
+            user.public_key = node_user["public_key"]
+
```

Comment:
The `is_unmessagable` flag should be included here.

```suggestion

    if "isUnmessagable" in node_user:
        user.is_unmessagable = node_user["isUnmessagable"]
```